arches_lingo/src/arches_lingo/App.vue

```diff
@@ -18,36 +25,49 @@ const user = ref<User | null>(null);
 const setUser = (userToSet: User | null) => {
     user.value = userToSet;
 };
-provide(userKey, { user, setUser });
+provide(USER_KEY, { user, setUser });
 
 const router = useRouter();
 const route = useRoute();
+const toast = useToast();
+const { $gettext } = useGettext();
 
-const isAuthenticated = computed(() => {
-    return user.value && user.value.username !== "anonymous";
-});
+router.beforeEach(async (to, _from, next) => {
+    try {
+        let userData = user.value;
 
-router.beforeEach(async (to) => {
-    if (
-        !isAuthenticated.value &&
-        // Avoid an infinite redirect
-        to.name !== routeNames.login
-    ) {
-        return { name: routeNames.login, params: { next: to.name as string } };
+        if (!userData) {
+            userData = await fetchUser();
+            setUser(userData);
```

Comment:
Apologies if my last suggested change is what set this awry, but I'm now noticing that when I:
- start with a fresh browser (no cookies)
- navigate to root (result: I get a csrftoken cookie as expected)
- I log in
- nothing happens (network tab shows login succeeded, but i'm not navigated away. now the session cookie exists, though)

Maybe we need to check for the existence of the session cookie also before we short-circuit like this?

Reply:
...or check for auth'd user (not anonymous user)?

Reply:
no worries and good catch!